packages/starlight/user-components/Badge.astro

```diff
@@ -0,0 +1,139 @@
+---
+import { type Badge, BadgeComponentSchema } from '../schemas/badge';
+import { parseWithFriendlyErrors } from '../utils/error-map';
+
+type Props = Badge;
```

Comment:
At the moment, this is equivalent to `z.output<ReturnType<typeof badgeSchema>>` but I guess we would want the props to be `z.output<typeof BadgeComponentSchema>` instead so we may need to create a type for this new component schema.